sudoku.py

```diff
@@ -0,0 +1,453 @@
+            #         title=random.choice(NEGATIVE_REPLIES),
+            #         description="Uh oh! You took too long to respond!",
+            #         color=Colours.soft_red
+            #     )
+
+                # await ctx.send(ctx.author.mention, embed=timeout_embed)
+
+            # await self.start(ctx)
+        elif coord and digit.isnumeric() and -1 < int(digit) < 10 or digit.lower() == "x":
+            print(f"{coord=}, {digit=}")
+            await game.update_board(digit, coord)
+        else:
+            raise commands.BadArgument
+            # await ctx.send("Welcome to Sudoku! Type your guesses like so: `A1 1`")
+            # await self.start(ctx)
+            # await self.bot.wait_for(event="message")
+            # if coord and digit.isnumeric() and -1 < int(digit) < 10 or digit in "xX":
+            #     # print(f"{coord=}, {digit=}")
+            #     await game.update_board(digit, coord)
+            # else:
+            #     raise commands.BadArgument
+
+    @sudoku.command()
+    async def start(self, ctx: commands.Context) -> None:
+        """Start a Sudoku game."""
+        if self.games.get(ctx.author.id):
+            await ctx.send("You are already playing a game!")
+            return
+        self.games[ctx.author.id] = SudokuGame(ctx)
+        await ctx.send("Started a Sudoku game.")
+
+    @sudoku.command(aliases=["end", "stop"])
+    async def finish(self, ctx: commands.Context) -> None:
+        """End a Sudoku game."""
+        game = self.games.get(ctx.author.id)
+        if game:
+            if ctx.author == game.invoker:
+                del self.games[ctx.author.id]
+                await ctx.send("Ended the current game.")
+            else:
+                await ctx.send("Only the owner of the game can end it!")
+        else:
+            await ctx.send("You are not playing a Sudoku game! Type `.sudoku start` to begin.")
+
+    @sudoku.command(aliases=["who", "information", "score"])
+    async def info(self, ctx: commands.Context) -> None:
+        """Send info about a currently running Sudoku game."""
+        game = self.games.get(ctx.author.id)
+        if game:
+            await ctx.send(embed=game.info_embed())
+        else:
+            await ctx.send("You are not playing a game!")
+
+    @sudoku.command()
+    async def hint(self, ctx: commands.Context) -> None:
+        """Fill in one empty square on the Sudoku board."""
+        game = self.games.get(ctx.author.id)
+        if game:
+            game.hints.append(time.time())
+            while True:
+                x, y = random.randint(0, 5), random.randint(0, 5)
+                if game.puzzle[x][y] == 0:
+                    await game.update_board(digit=random.randint(0, 5), coord=(x, y))
+                    break
+
+
+class SudokuView(discord.ui.View):
+    """A set of buttons to control a Sudoku game."""
+
+    def __init__(self, ctx: commands.Context):
+        super(SudokuView, self).__init__()
```

Comment:
Don't worry, that is Python 2 syntax. You can do this now:

```suggestion
        super().__init__()
```

Reply:
Ah ok, again not me who wrote that 😅